src/hotspot/os/linux/os_linux.cpp

```diff
@@ -1786,6 +1786,18 @@ void * os::Linux::dll_load_in_vmthread(const char *filename, char *ebuf,
   return result;
 }
 
+const char* os::Linux::dll_path(void* lib) {
+  struct link_map *lmap;
+  const char* l_path = NULL;
+  if (lib != NULL) {
```

Comment:
I still say this can just be an assert - it is checked in the callers.

Reply:
I introduced an assert in dll_path.